### DISCUSSION TITLE
Aggregating pivot table attributes in overview.

### DISCUSSION THREAD

**Discussion by kolaente** [filamentphp/filament#6267] [Help] [answered]
In my app, I have a `Box` model which has a many-to-many relation with `Items`. This relation has `count` attribute in the pivot table.

In the Filament overview, I now want to show a column with the total count of all related items. Basically, this sql query:

```sql
select boxes.*, sum(box_item.count) as total_items
from `boxes`
         left join `box_item` on `box_item`.`box_id` = `boxes`.`id`
group by `boxes`.`id`;
```

As per [the docs](https://filamentphp.com/docs/2.x/tables/columns/relationships#aggregating-relationships) I tried to use a `->sum('items', 'box_item.count')` but that seems to only work for aggregations on attributes on the related entity (which would be `Item` in this case and not what I want).

I then tried to adjust the eloquent query in the BoxResource:

```php
public static function getEloquentQuery(): Builder
{
    return parent::getEloquentQuery()
        ->select(DB::raw('boxes.*, sum(box_item.count) as total_items'))
        ->leftJoin('box_item', 'box_item.box_id', 'boxes.id')
        ->groupBy('boxes.id');
}
```

But that seems to fail during the pagination query with an sql error:

```
SQLSTATE[42000]: Syntax error or access violation: 1055 'zool.boxes.title' isn't in GROUP BY (Connection: mysql, SQL: select count(*) as aggregate from (select boxes.*, sum(box_item.count) as total_items, (select sum(`box_item`.`count`) from `items` inner join `box_item` on `items`.`id` = `box_item`.`item_id` where `boxes`.`id` = `box_item`.`box_id`) as `items_sum_box_itemcount` from `boxes` left join `box_item` on `box_item`.`box_id` = `boxes`.`id` group by `boxes`.`id`) as `aggregate_table`)
```

Any idea how to solve this?

**Comment by danharrin** [comment, marked as answer]
Including a group by in an Eloquent query prevents Eloquent from being able to properly hydrate the models into an Eloquent collection. Because of this, it won't work in a Filament table

**Reply by kolaente** [reply]
Okay, but there's no way to show the count? I could probably do a computed attribute on the item, but I feel like that would be really inefficient. There's got to be a better way.

**Reply by danharrin** [reply]
You can do a select with the count, you just can't group

**Reply by kolaente** [reply]
But because of the count that query won't work without a `group by`:

```
SQLSTATE[42000]: Syntax error or access violation: 1140 Mixing of GROUP columns (MIN(),MAX(),COUNT(),...) with no GROUP columns is illegal if there is no GROUP BY clause (Connection: mysql, SQL: select boxes.*, sum(box_item.count) as total_items from `boxes` left join `box_item` on `box_item`.`box_id` = `boxes`.`id` limit 10 offset 0)

**Comment by micraux** [comment]
Hello @kolaente,

I think I found a solution: at least it works for my use case...

The idea is to create a function in your model, then call the function in your resource.

In your BOX model:
```php
public function total_items()
{
    return $this->items()->sum('quantity');
}
```

In your BOX resource:
```php
Tables\Columns\TextColumn::make('total_items')
    ->state(
        fn (Box $record): int => $record->total_items()
    )
```
